libafl_libfuzzer/libafl_libfuzzer_runtime/src/fuzz.rs

```diff
@@ -0,0 +1,333 @@
+    // in the future, we can handle this explicitly, but this introduces issues with generics :(
+
+    // A minimization+queue policy to get testcasess from the corpus
+    let scheduler =
+        IndexesLenTimeMinimizerScheduler::new(PowerQueueScheduler::new(PowerSchedule::FAST));
+
+    // A fuzzer with feedbacks and a corpus scheduler
+    let mut fuzzer = StdFuzzer::new(scheduler, feedback, objective);
+
+    // The wrapped harness function, calling out to the LLVM-style harness
+    let mut harness = |input: &BytesInput| {
+        let target = input.target_bytes();
+        let buf = target.as_slice();
+        harness(buf.as_ptr(), buf.len());
+        ExitKind::Ok
+    };
+
+    let mut tracing_harness = harness;
+
+    // Create the executor for an in-process function with one observer for edge coverage and one for the execution time
+    let mut executor = TimeoutExecutor::new(
+        InProcessExecutor::new(
+            &mut harness,
+            tuple_list!(edges_observer, time_observer, backtrace_observer),
+            &mut fuzzer,
+            &mut state,
+            &mut mgr,
+        )?,
+        options.timeout(),
+    );
+
+    // TODO Setup a tracing stage in which we log comparisons
+    let tracing = TracingStage::new(InProcessExecutor::new(
+        &mut tracing_harness,
+        tuple_list!(cmplog_observer),
+        &mut fuzzer,
+        &mut state,
+        &mut mgr,
+    )?);
+
+    // The order of the stages matter!
+    let mut stages = tuple_list!(
+        calibration,
+        tracing,
+        i2s,
+        cm_i2s,
+        std_power,
+        cm_power,
+        cm_std_power,
+        cc_std_power,
+        cc_power,
+    );
+
+    // In case the corpus is empty (on first run), reset
+    if state.corpus().count() < 1 {
+        if !options.dirs().is_empty() {
+            println!("Loading from {:?}", options.dirs());
+            // Load from disk
+            state
+                .load_initial_inputs(&mut fuzzer, &mut executor, &mut mgr, options.dirs())
+                .unwrap_or_else(|_| {
+                    panic!("Failed to load initial corpus at {:?}", options.dirs())
+                });
+            println!("We imported {} inputs from disk.", state.corpus().count());
+        }
+        if state.corpus().count() < 1 {
+            // Generator of printable bytearrays of max size 32
+            let mut generator = RandBytesGenerator::from(RandBytesGenerator::new(32));
+
+            // Generate 8 initial inputs
+            state
+                .generate_initial_inputs(&mut fuzzer, &mut executor, &mut generator, &mut mgr, 8)
+                .expect("Failed to generate the initial corpus");
+            println!(
+                "We imported {} inputs from the generator.",
+                state.corpus().count()
+            );
+        }
+    }
+
+    fuzzer.fuzz_loop(&mut stages, &mut executor, &mut state, &mut mgr)?;
```

Comment:
We can just use Launcher for fork mode, here, right

Reply:
Also, we'd want to use the same for -jobs -- we don't support (and probably can't support) multithreaded but same process.

Reply:
Without threading in libfuzzer, it simply stops on the first crash - so for this we should use the simple manager instead.
Maybe that'll need two very similar functions then..

Reply:
I figured out how to do it with a closure instead :grin:

Reply:
Would be awesome to somehow add a simple manager replay to the launcher as well